api/model/storage/postgres/timeseries.go

```diff
@@ -262,27 +290,78 @@ func (f *TimeSeriesField) fetchRepresentationTimeSeries(categoryBuckets []*api.B
 	return timeseriesExemplars, nil
 }
 
+func (s *Storage) fetchTimeSet(dataset string, storageName string, xColName string) (pgx.Rows, error) {
+	query := fmt.Sprintf("SELECT ARRAY_AGG( DISTINCT \"%s\" order by \"%s\") FROM %s", xColName, xColName, storageName)
+	res, err := s.client.Query(query)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to fetch timeseries set from postgres")
+	}
+	return res, nil
+}
+
+func (s *Storage) parseDateTimeSet(rows pgx.Rows) (*map[time.Time]float64, *[]time.Time, error) {
+	result := map[time.Time]float64{}
+	if rows != nil {
+		defer rows.Close()
+	}
+	timeArr := []time.Time{}
+	rows.Next() // should only be 1 row
```

Comment:
May want to put in place an if to make sure there's a row. It could lead to a more graceful error.